policyengine_us/variables/gov/states/md/tax/income/credits/ctc/md_ctc.py

```diff
@@ -24,8 +26,33 @@ def formula_2020(tax_unit, period, parameters):
         meets_age_limit = person("age", period) < age_limit
         eligible = dependent & meets_age_limit
         eligible_children = tax_unit.sum(eligible)
-        md_ctc = eligible_children * p.amount
+
+        # Calculate base credit amount
+        base_credit = eligible_children * p.amount
+
+        # Apply federal CTC reduction if applicable
         if p.reduced_by_federal_credit:
             federal_ctc = tax_unit("ctc", period)
-            return max_(md_ctc - federal_ctc, 0)
-        return md_ctc
+            base_credit = max_(base_credit - federal_ctc, 0)
+
+        # Apply income-based phase-out instead of hard cutoff starting from specified year
+        if period.start.year >= p.phase_out_start_year:
+            agi = tax_unit("adjusted_gross_income", period)
+            phase_out_threshold = p.phase_out_threshold
+            phase_out_rate = p.phase_out_rate
+            phase_out_increment = p.phase_out_increment
+
+            # Calculate phase-out reduction: reduction for every increment (or fraction thereof) above threshold
+            excess_income = max_(agi - phase_out_threshold, 0)
+            # Use ceiling division: any fraction of increment counts as a full increment for phase-out
+            increments_above_threshold = ceil(
+                excess_income / phase_out_increment
+            )
+            phase_out_reduction = increments_above_threshold * phase_out_rate
+
+            # Apply phase-out (credit cannot go below zero)
+            return max_(base_credit - phase_out_reduction, 0)
+
+        # For years before phase-out start year, use old eligibility logic (hard cutoff)
+        agi_eligible = tax_unit("adjusted_gross_income", period) <= p.agi_cap
+        return where(agi_eligible, base_credit, 0)
```

Comment:
This is adjusting pre-2025 code -- dud we have a bug before?

Reply:
We should then be adjusting `md_ctc_eligible` if we were using the wrong AGI concept before -- we should also only have one AGI check in `md_ctc_eligible` and **not** in `md_ctc` .

Not clear why we need to check for AGI twice

Also please use `defined_for =  "md_ctc_eligible"` in `md_ctc` -- see diff below 

![image](https://github.com/user-attachments/assets/13357afe-7697-4332-8280-19f23bc9c9ae)